schematic/manifest/generator.py

```diff
@@ -1677,16 +1682,25 @@ def create_manifests(
                     "Please check your submission and try again."
                 )
 
-        data_model_parser = DataModelParser(path_to_data_model=path_to_data_model)
+        if graph_data_model is None:
```

Comment:
so the logic from 1685 to 1703 can be further improved. Essentially you want to cover the following cases: 
1) if graph data model is provided, then use the graph data model
2) if the pickle file is provided, get the graph from the pickle file and use that
3) if none of these two are provided, parse data model to get the graph. 

so instead of starting with `if graph_data_mode is None`, it will be easier if you start with if graph_data_model

and also, please consider wrapping this part in its own function... that way testing can be easier